src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java

```diff
@@ -230,16 +233,38 @@ else if (!isCharacterValidAfterGenericEnd(charAfter)) {
     }
 
     /**
-     * Checks if generic is before constructor invocation.
+     * Checks if generic is before constructor invocation. Identifies two cases:
+     * <ol>
+     *     <li>{@code new ArrayList<>();}</li>
+     *     <li>{@code new Outer.Inner<>();}</li>
+     * </ol>
      *
      * @param ast ast
-     * @return true if generic before a constructor invocation
+     * @return true if generic is before constructor invocation
      */
-    private static boolean isGenericBeforeCtor(DetailAST ast) {
+    private static boolean isGenericBeforeCtorInvocation(DetailAST ast) {
+        final DetailAST grandParent = ast.getParent().getParent();
+        return grandParent.getType() == TokenTypes.LITERAL_NEW
+                || grandParent.getParent().getType() == TokenTypes.LITERAL_NEW;
+    }
+
+    /**
+     * Checks if generic is after {@code LITERAL_NEW}. Identifies three cases:
+     * <ol>
+     *     <li>{@code new <String>Object();}</li>
+     *     <li>{@code new <String>Outer.Inner();}</li>
+     *     <li>{@code new <@A Outer>@B Inner();}</li>
+     * </ol>
+     *
+     * @param ast ast
+     * @return true if generic after {@code LITERAL_NEW}
+     */
+    private static boolean isGenericAfterNew(DetailAST ast) {
```

Comment:
Method name change from `isGenericBeforeCtor()` to `isGenericAfterNew()`, this describes method's behaviour more accurately, as it only identified the cases where generic existed after `new` keyword.
The method didn't identified cases where generic preceded constructor invocation `()` which was the cause for the false negative.

The method now also identifies a new similar case for Annotations (case 3.)